week-5/C-promises/exercise.js

```diff
@@ -0,0 +1,132 @@
+/*
+  Promises
+  ---------------------------------
+  Open index.html in your browser
+*/
+
+/*
+  EXERCISE 1
+  =======
+
+  Insert the result of promise1 into the #exercise1 element. Hint: .then()
+*/
+function exercise1() {
+  var promise1 = resolvedPromise()
+    promise1.then(function(value) {
+    document.querySelector("#exercise1").innerText= `success:  ${value}`;
+ });
+}
+/*
+  EXERCISE 2
+  =======
+
+  Insert the result of promise2 into the #exercise2 element. Hint: .catch()
+*/
+function exercise2() {
+  var promise2 = rejectedPromise();
+  promise2.catch(function(value){
+  document.querySelector("#exercise2").innerText= `fail:  ${value}`;
+ })
+}
+/*
+  EXERCISE 3
+  =======
+
+  Insert the result of promise3 into the #exercise3 element. Notice how it
+  takes longer than the others to finish
+*/
+function exercise3() {
+  var promise3 = delayedPromise();
+  
+  promise3.then(function(value){
+    document.querySelector("#exercise3").innerText=`Delay: ${value}`;
+  })
+}
+
+/*
+  EXERCISE 4
+  =======
+
+  Take the result of promise4 and concatenate your name onto the end of it.
+  Then return your concatenated string as a promise. Finally insert the result
+  of your promise and insert it into the #exercise4 element
+*/
+function exercise4() {
+  var promise4 = concatPromise();
+  promise4.then(function(value){
+    document.querySelector("#exercise4").innerText=`${value}  Hasanein`;
+  })
+}
+
+/*
+  EXERCISE 5 (Stretch Goal)
+  =======
+  Start this exercise if you have completed all of the other exercises and
+  you understand them
+
+  Create a new Promise that resolves with the string "Hello Promises!".
+  Take the result of your promise and insert the it into the #exercise5
+  element
+*/
+
+function exercise5() {
+  // Write your implementation here
+  var promise5 = resolvedPromise();
```

Comment:
This isn't quite correct. The exercise asked you to create your own Promise, not use one that is provided for you. How can you create your own promises?